app/src/main/java/com/vero/tasky/agenda/data/repository/TaskRepositoryImpl.kt

```diff
@@ -0,0 +1,80 @@
+package com.vero.tasky.agenda.data.repository
+
+import com.vero.tasky.agenda.data.local.dao.ModifiedAgendaItemDao
+import com.vero.tasky.agenda.data.local.dao.TaskDao
+import com.vero.tasky.agenda.data.local.entities.ModifiedAgendaItemEntity
+import com.vero.tasky.agenda.data.mappers.toTask
+import com.vero.tasky.agenda.data.mappers.toTaskEntity
+import com.vero.tasky.agenda.data.remote.network.api.TaskApi
+import com.vero.tasky.agenda.domain.model.AgendaItem
+import com.vero.tasky.agenda.domain.model.AgendaItemType
+import com.vero.tasky.agenda.domain.model.ModificationType
+import com.vero.tasky.agenda.domain.remindermanager.AlarmData
+import com.vero.tasky.agenda.domain.remindermanager.AlarmHandler
+import com.vero.tasky.agenda.domain.repository.TaskRepository
+import com.vero.tasky.agenda.domain.usecase.SyncAgendaUseCase
+import com.vero.tasky.agenda.domain.workmanagerrunner.SaveTaskRunner
+import com.vero.tasky.core.data.remote.safeSuspendCall
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.mapNotNull
+
+class TaskRepositoryImpl(
+    private val api: TaskApi,
+    private val taskDao: TaskDao,
+    private val saveTaskRunner: SaveTaskRunner,
```

Comment:
You dont need a worker for the Task

Since Reminder and Task get saved instantly (you are just sending a json), theres no need for a WorkManager

On events, since you are uploading a photo (which might take a while) then it makes sense to have a WorkManager there

Reply:
Then I will need to write an additional Worker for syncing Agenda (tasks/reminders at a start) if the user uses the app offline.

Reply:
Yes exactly

If a coroutine fails, then you save the item locally (task and reminders)

And then have a worker that syncs